src/Views/Form/Templates/Sequoia/views/receipt.php

```diff
@@ -28,176 +32,61 @@
-					</div>
-				<?php endif; ?>
-			</div>
+		<?php require 'social-sharing.php'; ?>
+		<?php
+		/* @global DetailGroup $group */
+		foreach ( $receiptDetails->getDetailGroupList() as $detailGroupClassName ) {
+			$group = $receiptDetails->getDetailGroupObject( $detailGroupClassName );
 
-			<!-- Payment Details -->
-			<div class="details-table payment-details">
-				<div class="details-row">
-					<div class="detail">
-						<?php _e( 'Payment Method', 'give' ); ?>
-					</div>
-					<div class="value">
-						<?php echo getGatewayLabel( $donation->gateway ); ?>
-					</div>
-				</div>
-				<div class="details-row">
-					<div class="detail">
-						<?php _e( 'Payment Status', 'give' ); ?>
-					</div>
-					<div class="value">
-						<?php echo getDonationStatusLabel( $donation->ID, true ); ?>
-					</div>
-				</div>
-				<div class="details-row">
-					<div class="detail">
-						<?php _e( 'Donation Amount', 'give' ); ?>
-					</div>
-					<div class="value">
-						<?php
-						echo filterCurrency(
-							sanitizeAmount( $donation->subtotal ),
-							[
-								'currency_code'   => $donation->currency,
-								'decode_currency' => true,
-								'form_id'         => $donation->form_id,
-							]
-						);
-						?>
-					</div>
-				</div>
-				<div class="details-row">
-					<div class="detail">
-						<?php _e( 'Processing Fees', 'give' ); ?>
-					</div>
-					<div class="value">
-						<?php
-						$fees = $donation->total - $donation->subtotal;
-						echo filterCurrency(
-							sanitizeAmount( $fees ),
-							[
-								'currency_code'   => $donation->currency,
-								'decode_currency' => true,
-								'form_id'         => $donation->form_id,
-							]
-						);
-						?>
-					</div>
-				</div>
-				<div class="details-row total">
-					<div class="detail">
-						<?php _e( 'Donation Total', 'give' ); ?>
-					</div>
-					<div class="value">
-						<?php
-						echo filterCurrency(
-							sanitizeAmount( $donation->total ),
-							[
-								'currency_code'   => $donation->currency,
-								'decode_currency' => true,
-								'form_id'         => $donation->form_id,
-							]
-						);
-						?>
-					</div>
-				</div>
-			</div>
-		</div>
+			if ( ! $group->canShow() ) {
+				continue;
+			}
+
+			echo '<div class="details">';
+			if ( $group->heading ) {
+				printf( '<h3 class="headline">%1$s</h3>', $group->heading );
+			}
+
+			if ( $detailList = $group->getDetailsList() ) {
+				echo '<div class="details-table">';
+
+				/* @var Detail $detail */
+				foreach ( $detailList as $detailItemClassName ) {
+					$detail = $group->getDetailItemObject( $detailItemClassName );
+					$value  = $detail->getValue();
+
+					if ( ! $value ) {
+						continue;
+					}
+
+					// This class is required to highlight total donation amount in receipt.
+					$detailRowClass = $detailItemClassName === TotalAmountDetailItem::class ? ' total' : '';
+
+					printf(
+						'<div class="details-row%1$s">',
+						$detailRowClass
```

Comment:
Needs to be escaped

Reply:
@bradp It is no user input, so no need to escape this. I am calculate this dynamically in this loop.